### PR TITLE
Add Big Bird to FARM

### PR DESCRIPTION
**Pull request by tanmaylaud**
requires latest version of transformers v4.7.0.dev0

### CODE REVIEW

**Comment by Timoeller** [comment]
Nice one, thanks @tanmaylaud 

We are planning a FARM release mid next week. Would be nice if we can get this in, but I would prefer not including the dev transformers version. Already testing current master in #791

**Comment by tanmaylaud** [comment]
> Nice one, thanks @tanmaylaud
> 
> We are planning a FARM release mid next week. Would be nice if we can get this in, but I would prefer not including the dev transformers version. Already testing current master in #791

Yes, may be we can merge it in a later release.

**Comment by julian-risch** [comment]
Hi @tanmaylaud FARM is no upgraded to transformers=4.7.0 and we are about to merge your Big Bird PR 🎉 Did you do some testing, e.g., running the [doc classification example](https://github.com/deepset-ai/FARM/blob/master/examples/doc_classification.py) with [a Big Bird model](https://huggingface.co/google/bigbird-roberta-base/tree/main)?

File: farm/modeling/language_model.py [hunk 1]
```diff
@@ -1614,3 +1619,99 @@ def enable_hidden_states_output(self):
...
+            big_bird.name = kwargs["farm_lm_name"]
+        else:
+            big_bird.name = pretrained_model_name_or_path
+        # We need to differentiate between loading model using FARM format and Pytorch-Transformers format
+        farm_lm_config = Path(pretrained_model_name_or_path) / "language_model_config.json"
+        if os.path.exists(farm_lm_config):
+            # FARM style
+            big_bird_config = BigBirdConfig.from_pretrained(farm_lm_config)
+            farm_lm_model = Path(pretrained_model_name_or_path) / "language_model.bin"
+            big_bird.model = BigBirdModel.from_pretrained(farm_lm_model, config=bert_config, **kwargs)
```
Comment (julian-risch) [inline farm/modeling/language_model.py:1671]: `bert_config` should be `big_bird_config`here. Otherwise `bert_config`is undefinded.

**Review by julian-risch — CHANGES_REQUESTED** [review]
One small change needed. Just a mix up of `bert_config` and `big_bird_config`. All other changes look very good to me. Ready to merge after this small change.
I tested the doc_classification example with google/bigbird-roberta-base.

File: farm/modeling/language_model.py (hunk 1 above)
Comment (tanmaylaud) [reply farm/modeling/language_model.py]: Thanks!

**Comment by tanmaylaud** [comment]
Regarding testing,
We did a dry run (training) on SQUAD:
https://public-mlflow.deepset.ai/#/experiments/2/runs/40a887dac3e24e75b891a0b050aba3dc

In order to full inspect the performance of bigbird, we need larger passages, which are not available in SQUAD.

**Review by julian-risch — APPROVED** [review]

**Pull request merged**